\documentclass[a4paper,11pt]{article}

\usepackage[latin1]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{bbm} %math chars
\usepackage{amsmath}
\usepackage{indentfirst}
\usepackage{fullpage} %minimizes the default margins
\usepackage{url}
\usepackage{graphicx}
\usepackage[center,footnotesize]{caption} %options des legendes des graphes
\usepackage[section]{placeins} %place les figures d'une section avant le debut de la suivante
\usepackage{subfig} %a) b) c)

\title{Series 2}
\date{September 27, 2011}
\author{Genomics and bioinformatics - Week 2}

\begin{document}
\maketitle

\section{Description}
In today's session you will use publicly available genome sequence and annotation data for a particular species
to extract some biological information about that species.


\section{Visualizing genome data online}

Go to the UCSC Genome Browser and select the $Mus$ $musculus$ genome.

Visualize the most recent assembly (mm9) of mouse chromosome 18.

Scroll down to "Mapping and Sequencing Tracks" and load the GC percent track. 

You can obtain more information by clicking on the track once it loads.  

\section{Downloading genome data for mouse chr18}
Copy the sequence file \texttt{chr18.fa} and annotation files \texttt{chr18.gtf}, \texttt{chr18\_mod.txt} from the USB keys.

\section{Programming exercise}

\subsection{Using Python}
\begin{enumerate}
\item Load the \texttt{.fa} file and extract the chromosome sequence
\item Determine the length of the sequence
\item Calculate the number of As, Gs, Cs and Ts in the sequence
\item Compute the GC-content of the chromosome
\item Plot GC content along the chromosome using "appropriate" window (bin) sizes (use  \texttt{matplotlib})
\item Write the start and end coordinates of each bin and it's corresponding GC content to a file, as follows,
\scriptsize\texttt {binStart binEnd	GC\_content}
\end{enumerate}

\subsection{Using R}
\subsubsection{Part I}

\normalsize\underline{Note}

\normalsize The \texttt{.gtf} file is a tab-delimited file, with the following column headers,

\scriptsize\texttt {chromosome	source	feature	start	end	score	strand	frame	attributes}

\begin{enumerate}
\normalsize\item Load the \texttt{.gtf} file for chr18 in R
\item Extract the rows corresponding to exons from the \texttt{feature} column to a new table
\item Compute exon sizes and attach a new column \texttt{exonSize} to the table
\item Plot the exon size distribution for chr 18
\end{enumerate}

\subsubsection{Part II}

\normalsize\underline{Note}

\normalsize The modified \texttt{.gtf} file, \texttt{chr18\_attributes.txt} is a tab-delimited file, with the following column headers,

\scriptsize\texttt {chromosome	source	feature	start	end	score	strand	frame	gene\_id	 transcript\_id 	 exon\_number 	 gene\_name 	 gene\_biotype 	 transcript\_name 	 protein\_id }

\begin{enumerate}
\normalsize\item Load the modified \texttt{.gtf} file for chr18 in R
\item Extract the rows corresponding to exons from the \texttt{feature} column to a new table
\item Make a table containing t
\item Plot the exon size distribution for chr 18
\end{enumerate}


\Large {\emph\bf Questions}
\begin{enumerate}
\item Comment on the exon size distribution for mouse chr 18
\normalsize\item List names of genes containing the,

 a) the longest exon, b) the shortest exon and c) most number of exons.
\item List all the intronless genes in the chromosome.
\end{enumerate}

\subsection{Reference documentation}
\normalsize For R - \url{http://cran.r-project.org/doc/manuals/refman.pdf}

For Python - \url{http://docs.python.org/tutorial/}\\

\emph{If you need help:}
\begin{enumerate}
\item Go through last week's exercise session for examples
\item Use Google
\item Use the ? or help() with R commands
\item Ask us
\end{enumerate}

\end{document}
