\documentclass[a4paper,11pt]{article}
\usepackage[latin1]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{bbm} %math chars
\usepackage{amsmath}
\usepackage{indentfirst}
\usepackage{fullpage} %minimizes the default margins
\usepackage{url}
\usepackage{graphicx}
\usepackage[center,footnotesize]{caption} %options des legendes des graphes
\usepackage[section]{placeins} %place les figures d'une section avant le debut de la suivante
\usepackage{subfig} %a) b) c)
\title{Series 2}
\date{September 27, 2011}
\author{Genomics and bioinformatics - Week 2}
\begin{document}
\maketitle


In today's session, you will use publicly available genome sequence and annotation data for a particular species
to extract some biological information about that species.

If you do not have a working copy of Python and R on your computer, please go through last week's tutorial before starting this exercise.



\section{Getting data with UCSC}
\subsection{Visualizing genome data}
\begin{enumerate}
\item Go to the UCSC Genome Browser and select the \textit{Mus musculus} genome.
\item Visualize the most recent assembly (mm10) of mouse chromosome 18.
\item Scroll down to ``Mapping and Sequencing Tracks'' and load the GC percent track.
\end{enumerate}
 You can obtain more information about the tracks by clicking on them.
\subsection{Downloading genome data}
Copy the ``sequence'' file \texttt{chr18.fa} and ``annotation'' files \texttt{chr18.gtf}, \texttt{chr18\_mod.txt} 
for mouse chr18 from the USB keys provided by us. 

Actually, the \texttt{.fa} files for mouse can be downloaded from the UCSC Downloads page:\\
\url{http://hgdownload.cse.ucsc.edu/goldenPath/mm9/chromosomes/},\\
and the .gtf file for the whole mouse genome can be downloaded from ENSEMBL:\\
\url{ftp://ftp.ensembl.org/pub/release-64/gtf/mus_musculus/} .



\section{Manipulating data with Python}
\begin{enumerate}
\item Load the \texttt{.fa} file for chr18 and extract the sequence.
\item Determine the length of the sequence.
\item Calculate the number of As, Gs, Cs and Ts in the sequence.
\item Compute the GC content of the chromosome.
\item Plot GC content along mouse chr18 using an appropriate window (bin) sizes (use \texttt{matplotlib}).
\item Write the start and end coordinates of each bin and it's corresponding GC content to a file, as follows: \\
\texttt {binStart <tab> binEnd <tab> GC\_content}
\end{enumerate}



\section{Manipulating data with R}
\subsection{Exons}
The \texttt{.gtf} file is a tab-delimited file, with the following column headers: \\
\texttt {chromosome source feature start end score strand frame attributes} .
\begin{enumerate}
\item Load the \texttt{.gtf} file for chr18 in R.
\item Extract the rows corresponding to exons from the \texttt{feature} column to a new table.
\item Compute exon sizes, and attach them to the table in a new column ``\texttt{exonSize}''.
\item Plot the exon size distribution for chr 18.
\end{enumerate}
\subsection{Genes}
The modified annotation file \texttt{chr18\_attributes.txt} is also a tab-delimited file, with the \indent following column headers: \\
\texttt {chromosome source feature start end score strand frame gene\_id transcript\_id exon\_number gene\_name gene\_biotype transcript\_name protein\_id } .
\begin{enumerate}
\normalsize
\item Load the modified \texttt{.txt} annotation file for chr18 in R.
\item Find out the ID and name of the gene containing,

 a) the longest exon, b) most number of exons.
\item List all the intron-less genes in the chromosome.
\end{enumerate}
\subsection{GC content}
\begin{enumerate}
\item Load the file (table) generated by your python script into R.
\item Recreate the GC content plot for chr18 in R.
\end{enumerate}

\end{document}