\documentclass[a4paper,11pt]{article}
\usepackage[latin1]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{bbm} %math chars
\usepackage{amsmath}
\usepackage{indentfirst}
\usepackage{fullpage} %minimizes the default margins
\usepackage{url}
\usepackage{graphicx}
\usepackage[center,footnotesize]{caption} %options des legendes des graphes
\usepackage[section]{placeins} %place les figures d'une section avant le debut de la suivante
\usepackage{subfig} %a) b) c)
\title{Series 2}
\date{}
\author{Genomics and bioinformatics - Week 3 - October 2, 2012}
\begin{document}
\maketitle

\section{Overlap graphs}
Here are five reads: TCTATA, GGTCTA, TATCTA, TCTAGC and TATATC. They originate from a single longer contig.
\begin{enumerate}
\item Try to reassemble the contig by hand, looking at the overlaps between the reads.
\item Build an Hamiltonian overlap graph with these reads and draw the longest path.
From there, reconstruct the contig.
\item Take $l=4$ and build the corresponding ``de Bruijn'' graph. Find an Eulerian cycle and reconstruct the contig. \\
\textit{Note}: The ``de Bruijn'' graph must be Eulerian, thus some edges of the graph will have to be doubled.
Try to understand which ones must be doubled (look at the overlaps between reads) and think about a criterion to do it automatically.
\end{enumerate}

\section{Getting data with UCSC}
\subsection{Visualizing genome data}
\begin{enumerate}
\item Go to the UCSC Genome Browser and select the C.Elegans genome (Nematode).
\item Visualize the most recent assembly (ce10) of chromosome I.
\item Scroll down to ``Mapping and Sequencing Tracks'' and load the GC percent track.
\end{enumerate}

\subsection{Downloading genome data and look at them}
Copy the ``sequence'' file \texttt{chrI.fa} and the ``annotation'' file \texttt{Caenorhabditis\_elegans.WS220.64.gtf} for C.Elegans from our USB key, or download them from UCSC and ENSEMBL: \\\\
\url{http://hgdownload.cse.ucsc.edu/goldenPath/ce10/chromosomes/}\\
\url{ftp://ftp.ensembl.org/pub/release-64/gtf/caenorhabditis\_elegans/}\\\\
Have a look at these two files with any text editor.


\newpage

\section{Manipulating data with Python}
\begin{enumerate}
\item Load the file \texttt{chrI.fa} and extract the sequence.
\item Determine the length of the sequence (see \texttt{len}).
\item Calculate the number of As, Gs, Cs and Ts in the sequence.
\item Compute the GC content of the sequence.
\item Plot the GC content using an appropriate window (bin) size (use \texttt{matplotlib}).
\item Write the start and end coordinates of each bin and its corresponding GC content to a file as follows:
\texttt {binStart <tab> binEnd <tab> GC\_content}
\end{enumerate}
\textit{Note}: If you feel confident, this is a good occasion to save time trying the Biopython library: \\
\texttt{from Bio import SeqIO} \# then use SeqIO.read()\\
\texttt{from Bio.SeqUtils import GC} \# then use GC() \\
\textit{Note}: list methods such as \texttt{count}, \texttt{append}, etc. may be useful.


\section{Manipulating data with R}
\subsection{Exons}
The file \texttt{Caenorhabditis\_elegans.WS220.64.gtf} is a tab-delimited file, with the following column headers: \\
\texttt {chromosome source feature start end score strand frame attributes} .
\begin{enumerate}
\item Load the file \texttt{Caenorhabditis\_elegans.WS220.64.gtf} in R. It may take a few minutes.
\item Extract the rows corresponding to exons from the \texttt{feature} column to a new table.
\item Compute the exon sizes, and attach them to the table in a new column ``\texttt{exonSize}''.
\item Plot the exon size distribution.
\end{enumerate}

\subsection{GC content}
\begin{enumerate}
\item Load the file (table) generated by your python script into R.
\item Recreate the GC content plot in R.
\end{enumerate}
\noindent
Note: Some useful features for this exercise:
\begin{itemize}
\item \texttt{which, length, hist}
\end{itemize}

\end{document}
